framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -889,11 +920,42 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
         } while ((retry_count < kRetryLimit) && capture_event->type != out_eventData->type);
         if (capture_event->type != out_eventData->type)
         {
-            GFXRECON_LOG_ERROR("Event %s (0x%x %d) never occurred!",
-                               GetEventTypeString(capture_event->type),
-                               capture_event->type,
-                               capture_event->type);
-            replay_result = XR_ERROR_RUNTIME_FAILURE; // Runtime never gave us the event we were looking for
+            if (IsHandledEventType(capture_event->type))
+            {
+                GFXRECON_LOG_ERROR("Event %s (0x%x %d) never occurred!",
+                                   GetEventTypeString(capture_event->type),
+                                   capture_event->type,
+                                   capture_event->type);
+
+                // Runtime never gave us the event we were looking for
+                replay_result = XR_ERROR_RUNTIME_FAILURE;
+            }
+            else
+            {
+                GFXRECON_LOG_WARNING("Unhandled Event %s (0x%x %d) never occurred, pretending everything is still fine",
```

Comment:
If this isn't a handled event type, should we even go through the timeout loop?  I guess since all events are the same size, we don't have to have been complied with a header that knows about them to capture and match on replay...

Reply:
Hmm.... IDK.  I could see waiting a few loops.  Because what if it does come in.  But maybe we should only loop like 3 additional times?   How important do you think it is?